src/modules/musig/tests_impl.h

```diff
@@ -1078,52 +965,292 @@ void musig_test_vectors(void) {
+                pk[0] = vec_pk[0];
+                pk[1] = vec_pk[1];
+                pk[2] = vec_pk[2];
                 has_second_pk = 1;
                 second_pk_idx = 1;
                 break;
             case 1:
                 /* [X3, X2, X1] */
                 n_pks = 3;
-                memcpy(pk_ser_tmp[2], pk_ser[0], sizeof(pk_ser_tmp[0]));
-                memcpy(pk_ser_tmp[1], pk_ser[1], sizeof(pk_ser_tmp[1]));
-                memcpy(pk_ser_tmp[0], pk_ser[2], sizeof(pk_ser_tmp[2]));
+                pk[2] = vec_pk[0];
+                pk[1] = vec_pk[1];
+                pk[0] = vec_pk[2];
                 has_second_pk = 1;
                 second_pk_idx = 1;
                 break;
             case 2:
                 /* [X1, X1, X1] */
                 n_pks = 3;
-                memcpy(pk_ser_tmp[0], pk_ser[0], sizeof(pk_ser_tmp[0]));
-                memcpy(pk_ser_tmp[1], pk_ser[0], sizeof(pk_ser_tmp[1]));
-                memcpy(pk_ser_tmp[2], pk_ser[0], sizeof(pk_ser_tmp[2]));
+                pk[0] = vec_pk[0];
+                pk[1] = vec_pk[0];
+                pk[2] = vec_pk[0];
                 has_second_pk = 0;
                 second_pk_idx = 0; /* unchecked */
                 break;
             case 3:
                 /* [X1, X1, X2, X2] */
                 n_pks = 4;
-                memcpy(pk_ser_tmp[0], pk_ser[0], sizeof(pk_ser_tmp[0]));
-                memcpy(pk_ser_tmp[1], pk_ser[0], sizeof(pk_ser_tmp[1]));
-                memcpy(pk_ser_tmp[2], pk_ser[1], sizeof(pk_ser_tmp[2]));
-                memcpy(pk_ser_tmp[3], pk_ser[1], sizeof(pk_ser_tmp[3]));
+                pk[0] = vec_pk[0];
+                pk[1] = vec_pk[0];
+                pk[2] = vec_pk[1];
+                pk[3] = vec_pk[1];
                 has_second_pk = 1;
                 second_pk_idx = 2; /* second_pk_idx = 3 is equally valid */
                 break;
             default:
                 CHECK(0);
         }
-        musig_test_vectors_helper(pk_ser_tmp, n_pks, combined_pk_expected[i], has_second_pk, second_pk_idx);
+        musig_test_vectors_keyagg_helper(pk, n_pks, agg_pk_expected[i], has_second_pk, second_pk_idx);
+    }
+}
+
+void musig_test_vectors_noncegen(void) {
+    enum { N = 3 };
+    secp256k1_scalar k[N][2];
+    const unsigned char k32_expected[N][2][32] = {
+        {
+            {
+                0x8D, 0xD0, 0x99, 0x51, 0x79, 0x50, 0x5E, 0xB1,
+                0x27, 0x3A, 0x07, 0x11, 0x58, 0x23, 0xC8, 0x6E,
+                0xF7, 0x14, 0x39, 0x0F, 0xDE, 0x2D, 0xEE, 0xB6,
+                0xF9, 0x31, 0x6A, 0xEE, 0xBE, 0x5C, 0x71, 0xFC,
+            },
+            {
+                0x73, 0x29, 0x2E, 0x47, 0x11, 0x34, 0x7D, 0xD3,
+                0x9E, 0x36, 0x05, 0xEE, 0xD6, 0x45, 0x65, 0x49,
+                0xB3, 0x0F, 0x3B, 0xC7, 0x16, 0x22, 0x5A, 0x18,
+                0x65, 0xBA, 0xE1, 0xD9, 0x84, 0xEF, 0xF8, 0x9D,
+            },
+        },
+        /* msg32 is NULL */
```

Comment:
Nit: With the changed ordering, it's now key32 that's NULL, right?

Reply:
yes, sorry